src/Service/Storage/Ef/EfTenantStorage.cs

```diff
@@ -196,6 +198,12 @@ public Task RemoveTokenKey(int keyId)
         return db.TokenKeys.Where(k => k.KeyId == keyId).ExecuteDeleteAsync();
     }
 
+    public Task RemoveExpiredTokenKeys(CancellationToken cancellationToken)
+    {
+        return db.TokenKeys.Where(x => (TimeProvider.GetUtcNow().DateTime - x.CreatedAt).TotalDays > 30)
```

Comment:
Only inject TimeProvider here if you don't use it anywhere else. Don't litter it all over the code.